File: DATHelper/app/src/main/java/com/kishanprao/datHelper/base/stateMachine/Failed.kt
package com.kishanprao.datHelper.base.stateMachine

import com.kishanprao.datHelper.base.Constants

class Failed(val connData: ConnData) : AbstractConnectionState() {
    companion object {
        private val TAG = Failed::class.java.simpleName
    }

    override fun performAction(action: Action): ConnectionState {
        return when (action) {
            is Action.OnFailureHandled -> {
                if (connData.code == Constants.ConnectionTermOk) {
                    Connected(connData)
                } else {
                    Disconnected()
                }
            }
            else -> {
                return super.performAction(action)
            }
        }
    }
}

File: Dragon Android Transfer/Dragon Android Transfer/Base/Wireless/StateMachine/Failed.swift
//
//  Failed.swift
//  DragonSocket
//
//  Created by Kishan P Rao on 08/11/18.
//  Copyright © 2018 Kishan P Rao. All rights reserved.
//

import Foundation

class Failed: AbstractConnectionState {
    let data: ConnectionData
    init(_ data: ConnectionData) {
        self.data = data
    }
    
    override func performAction(_ action: Action) -> ConnectionState {
        let state: ConnectionState
        switch action {
        case Action.OnFailureHandled():
            if (data.connectionStatus == Constants.ConnectionTermOk) {
                state = Connected(data)
            } else {
                state = Disconnected()
            }
        default:
            state = super.performAction(action)
        }
        return state
    }
}
